src/compiler/moduleNameResolver.ts

```diff
@@ -1754,8 +1753,10 @@ function tryResolveJSModuleWorker(moduleName: string, initialDir: string, host:
         /*conditions*/ undefined,
     );
 }
+
+// TODO: knip applies the internal marker to _all_ declarations, not just the one overload.

```

Comment:
This looks like a knip bug; if any overload is marked `@internal`, the entire thing is treated as unexportable.

Reply:
Not sure yet what's the best way to go about this, both technically and DX-wise.

For now we'll have to ignore it as it won't be trivial to change in Knip.